modules/percolator/src/main/java/org/elasticsearch/percolator/QueryAnalyzer.java

```diff
@@ -435,6 +467,16 @@ private static Result handleDisjunction(List<Query> disjunctions, int minimumSho
         }
     }
 
+    private static int highestBoost(Set<QueryExtraction> extractions, Map<String, Integer> boostFields) {
+        int highestBoost = 1;
```

Comment:
should we really return 1 if nothing was extracted? Let's return Integer.MIN_VALUE (or NEGATIVE_INFINITY once we are on floats or doubles) to make sure it won't be selected if another query could be extracted